### DISCUSSION TITLE
How do I determine my Teleport version?

### DISCUSSION THREAD

**Discussion by WilliamLoy** [gravitational/teleport#8163] [Q&A] [answered]
Can someone please tell me how to determine what version of Teleport I am running?

**Comment by deusxanima** [comment, marked as answer]
There are a couple of ways. 

1. If you're logged into the server, you can run `teleport version` and it will tell you the binary version.
2. If you're logged into the auth or into the cluster via `tsh`, you can run `tctl get <node|proxy|auth> | egrep -i 'hostname: |version' | grep -vi 'v2'` to get the list of all nodes/auth/proxy servers with the full binary version

**Reply by tukusejssirs** [reply]
Open two using `yq`: `tctl get node | yq 'select(.spec.hostname == "node_hostname") | .spec.version'` (`node_hostname` is actual hostname of a node). Or the same thing, but using a node UUID: `tctl get node | yq 'select(.metadata.name == "node_uuid") | .spec.version'` where `node_uuid` is actual node UUID.

@deusxanima, is there a way to access `tctl get` outside the cluster? I’d like to do it on my local machine (which is not added to the TP cluster) after I logged in to Teleport. I know I could do it via SSH (e.g. `tsh ssh -t $username@$cluster $cmd`), however, I wish I could do it directly via `tctl`. :thinking:

**Reply by deusxanima** [reply]
Once you're logged into the cluster via `tsh login` you should be able to run the `tctl get` commands directly from your localhost without needing to log into the remote host. If you've downloaded the full teleport binary instead of just the tctl/tsh binaries to your local machine, make sure that it didn't accidentally create an empty `/var/lib/teleport` directory as tctl will try to use this instead of the remote cluster connection, b/c it assumes you're running `tctl` on an instance that's joined to Teleport if that dir is present.

If you're not able to get it working, please paste the error you're seeing when you get a moment and I'll be happy to review
